### PR TITLE
chore: allow CLI to use `endpointId` if it is available in package.json

### LINKED ISSUE #165: [TRI-583] CLI Init: If there's an endpoint ID in package.json, use it
<sub>From [SyncLinear.com](https://synclinear.com) | [TRI-583](https://linear.app/triggerdotdev/issue/TRI-583/cli-init-if-theres-an-endpoint-id-in-packagejson-use-it)</sub>

### PR DESCRIPTION
**Pull request by kaf-lamed-beyt**
fixes #165 

@ericallam kindly take a look at this and let me know if I'm on the right track.

/claim #165

### CODE REVIEW

File: packages/cli/src/cli/index.ts [hunk 1]
```diff
@@ -34,6 +38,9 @@ program
   )
   .version(getVersion(), "-v, --version", "Display the version number")
   .action(async (options) => {
+    if (endpointId) {
+      options.endpointId = endpointId;
+    }
     await initCommand(options);
```
Comment (ericallam) [inline packages/cli/src/cli/index.ts]: Can you move this code into the `initCommand` (specifically here: https://github.com/triggerdotdev/trigger.dev/blob/2d722b94bd448e39506f0d6c9fff3c83e373ef2b/packages/cli/src/commands/init.ts#L203), and please make use of `readJSONFile` from `utils/fileSystem`. Also make sure to use the `resolvedPath`

File: apps/webapp/app/presenters/JobPresenter.server.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { User } from "~/models/user.server";
```
Comment (ericallam) [inline apps/webapp/app/presenters/JobPresenter.server.ts]: This looks to be code from another PR, could you remove all this and only leave the changes for the CLI?

**Review by ericallam — CHANGES_REQUESTED** [review]

File: apps/webapp/app/presenters/JobPresenter.server.ts (hunk 2 above)
Comment (kaf-lamed-beyt) [reply apps/webapp/app/presenters/JobPresenter.server.ts]: Okay. I'll tend to it.

File: packages/cli/src/cli/index.ts (hunk 1 above)
Comment (kaf-lamed-beyt) [reply packages/cli/src/cli/index.ts]: Awesome! I'll do this too.

Comment (kaf-lamed-beyt) [reply packages/cli/src/cli/index.ts]: How does this look? 

```ts
    if (!options.endpointSlug) {
      resolvedOptions.endpointSlug = await promptEndpointSlug(path);
      const resolvedPath = resolvePath(options.projectPath);

      const packageJSONPath = pathModule.join(resolvedPath, "package.json");
      const packageJSON = await readJSONFile(packageJSONPath);

      if (packageJSON && packageJSON.endpointId) {
        options.endpointSlug = packageJSON.endpointId;
      }
    }
   ```

Comment (ericallam) [reply packages/cli/src/cli/index.ts]: Close! You should look for an endpointSlug before doing the `promptEndpointSlug` call, and then skip the`promptEndpointSlug` call if the endpointId is found in the package.json

Comment (kaf-lamed-beyt) [reply packages/cli/src/cli/index.ts]: Aha! Great! I'll tkae that into consideration now. 

Just to digress a bit. Have you had a look at #170? I added some changes a while ago.

Comment (ericallam) [reply packages/cli/src/cli/index.ts]: Just left another review on #170 👍

**Comment by kaf-lamed-beyt** [comment]
I pushed some changes a while ago. Kindly review @ericallam, thank you!

File: packages/cli/src/commands/init.ts [hunk 3]
```diff
@@ -201,7 +201,15 @@ const resolveOptionsWithPrompts = async (
     }
 
     if (!options.endpointSlug) {
-      resolvedOptions.endpointSlug = await promptEndpointSlug(path);
+      const resolvedPath = resolvePath(options.projectPath);
+      const packageJSONPath = pathModule.join(resolvedPath, "package.json");
+      const packageJSON = await readJSONFile(packageJSONPath);
+
+      if (packageJSON && packageJSON.endpointId) {
```
Comment (ericallam) [inline packages/cli/src/commands/init.ts]: Just realized that the `endpointId` needs to be looked for inside of `trigger.dev`. So `package['trigger.dev'].endpointId`

File: docker-setup.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
+## Setting up Docker for the first time.
```
Comment (ericallam) [inline docker-setup.md]: Can you remove this file from the PR?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -13,6 +13,7 @@ requests should be made against. The changes on the `main`
 branch are tagged into a release monthly.
```
Comment (ericallam) [inline CONTRIBUTING.md]: Can you remove these changes from the PR?

**Review by ericallam — CHANGES_REQUESTED** [review]

File: packages/cli/src/commands/init.ts (hunk 3 above)
Comment (kaf-lamed-beyt) [reply packages/cli/src/commands/init.ts]: Ah! 

So does that mean the `init` command of the CLI is only for trigger.dev? 

Not for any other project that uses it?

**Comment by kaf-lamed-beyt** [comment]
Added the suggested changes just now, kindly review @ericallam

File: packages/cli/src/commands/init.ts [hunk 6]
```diff
@@ -201,7 +201,15 @@ const resolveOptionsWithPrompts = async (
     }
 
     if (!options.endpointSlug) {
-      resolvedOptions.endpointSlug = await promptEndpointSlug(path);
+      const resolvedPath = resolvePath(options.projectPath);
+      const packageJSONPath = pathModule.join(resolvedPath, "package.json");
+      const packageJSON = await readJSONFile(packageJSONPath);
+
+      if (packageJSON && packageJSON["trigger.dev"].endpointId) {
```
Comment (ericallam) [inline packages/cli/src/commands/init.ts]: This could throw an error if `packageJSON["trigger.dev"]`, could probably just rewrite this to 

```ts
if (packageJSON && packageJSON["trigger.dev"] && packageJSON["trigger.dev"].endpointId) {
```

**Review by ericallam — CHANGES_REQUESTED** [review]

File: packages/cli/src/commands/init.ts (hunk 6 above)
Comment (kaf-lamed-beyt) [reply packages/cli/src/commands/init.ts]: I'll push a change now.

**Review by ericallam — APPROVED** [review]

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -21,60 +22,63 @@ branch are tagged into a release monthly.
...
+9. Run the app. See the section below.
 
 ## Running
 
 1. You can run the app with:
-      ```
-      pnpm run dev --filter webapp
-      ```
-      
-      It should run on port `3030`: [http://localhost:3030](http://localhost:3030/)
```
Comment (ericallam) [inline CONTRIBUTING.md:77]: These changes still need to be removed from this PR

File: packages/cli/src/commands/init.ts [hunk 8]
```diff
@@ -201,7 +201,19 @@ const resolveOptionsWithPrompts = async (
     }
 
     if (!options.endpointSlug) {
-      resolvedOptions.endpointSlug = await promptEndpointSlug(path);
+      const resolvedPath = resolvePath(options.projectPath);
+      const packageJSONPath = pathModule.join(resolvedPath, "package.json");
```
Comment (ericallam) [inline packages/cli/src/commands/init.ts]: This looks good! Almost ready to merge, just need to use the passed in `path` argument to the `resolveOptionsWithPrompts` function as it's already the resolvedPath, so there is no need to call `resolvePath` again.

**Review by ericallam — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 7 above)
Comment (kaf-lamed-beyt) [reply CONTRIBUTING.md:77]: Oh! The whole contributing.md file?

File: packages/cli/src/commands/init.ts (hunk 8 above)
Comment (kaf-lamed-beyt) [reply packages/cli/src/commands/init.ts]: Awesome! I'll tend to this now.

Comment (kaf-lamed-beyt) [reply packages/cli/src/commands/init.ts]: This was nice to work on. I had to read your feedback again, to properly grasp what you were trying to say.

Comment (kaf-lamed-beyt) [reply packages/cli/src/commands/init.ts]: I realized that this PR has a conflict. So I synced my forked repo with the main one and pulled. 

When I did `git status` I saw that a lot of files have been modified/created. Should I just abandon this PR and create a new one? 

image

cc: @ericallam

Comment (ericallam) [reply packages/cli/src/commands/init.ts]: Yea might be easiest just to create a new one 👍
